pyccel/ast/builtins.py

```diff
@@ -1156,7 +1156,7 @@ def __init__(self, *args):
             if lengths:
                 self._length = min(lengths)
             else:
-                self._length = self.args[0].shape[0]
+                self._length = PythonMin([a.shape[0] for a in self.args])
```

Comment:
Does `PythonMin` have a `simplify` argument? It would be nice to get a literal if all the shapes are known

Reply:
this case deals with when the shape is not known, I think if all the shapes are known, then it would fall under the first case